### PR TITLE
refactor(azure): refactor azure provider

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Refactor code from Azure provider to ensure that it can be used as a library from the SDK.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/azure_provider.py [hunk 1]
```diff
@@ -446,31 +446,8 @@ def test_connection(
...
-                raise validation_error
-            return Connection(error=validation_error)
-
-        except SystemExit as exit_error:
-            logger.error(
-                f"{exit_error.__class__.__name__}[{exit_error.__traceback__.tb_lineno}]: {exit_error}"
-            )
-            if raise_on_exception:
-                raise exit_error
-            return Connection(error=exit_error)
```
Comment (MrCloudSec) [inline prowler/providers/azure/azure_provider.py]: Why are you deleting this?

File: prowler/providers/azure/azure_provider.py [hunk 2]
```diff
@@ -76,24 +75,28 @@ class AzureProvider(Provider):
     # TODO: this is not optional, enforce for all providers
     audit_metadata: Audit_Metadata
 
-    def __init__(self, arguments):
+    def __init__(
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py:78]: do we need default values here? I think so, just to cover our back.

File: prowler/providers/azure/azure_provider.py [hunk 3]
```diff
@@ -500,7 +501,7 @@ def check_service_principal_creds_env_vars():
                 logger.critical(
                     f"Azure provider: Missing environment variable {env_var} needed to authenticate against Azure"
                 )
-                sys.exit(1)
+                raise SystemExit
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py:507]: Add a TODO here to replace it with our custom exceptions once they are ready.

File: prowler/providers/azure/azure_provider.py [hunk 4]
```diff
@@ -615,7 +616,7 @@ async def get_azure_identity():
                 logger.critical(
                     "It was not possible to retrieve any subscriptions, please check your permission assignments"
                 )
-                sys.exit(1)
+                raise SystemExit
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py:623]: Add a TODO here to replace it with our custom exceptions once they are ready.

File: prowler/providers/azure/azure_provider.py [hunk 5]
```diff
@@ -628,7 +629,7 @@ async def get_azure_identity():
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
             )
-            sys.exit(1)
+            raise error
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py:637]: Add a TODO here to replace it with our custom exceptions once they are ready.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

**Review by jfagoagas — COMMENTED** [review]
Remember also to modify the providers/common file to init the Azure provider there.

File: prowler/providers/azure/azure_provider.py (hunk 5 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py:637]: Done

File: prowler/providers/azure/azure_provider.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py:623]: Done

File: prowler/providers/azure/azure_provider.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py:507]: Done

**Review by jfagoagas — DISMISSED** [review]
👏

File: prowler/providers/azure/azure_provider.py (hunk 1 above)
Comment (jfagoagas) [reply prowler/providers/azure/azure_provider.py]: Deleted by accident and then restored.

File: prowler/providers/common/provider.py [hunk 6]
```diff
@@ -215,7 +219,15 @@ def set_global_provider(arguments):
...
+                elif "kubernetes" in provider_class_name.lower():
+                    global_provider = provider_class(
+                        arguments.kubeconfig_file,
+                        arguments.context,
+                        arguments.namespace,
+                        arguments.config_file,
+                        arguments.fixer_config,
+                    )
+                 else:
                     global_provider = provider_class(arguments)
```
Comment (MrCloudSec) [inline prowler/providers/common/provider.py]:
```suggestion
```

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**